### PR TITLE
[JS] Fix video not autoplaying on iOS Safari

### PR DESCRIPTION
**Pull request by dclaux**
## Related Issue
Fixes https://github.com/microsoft/AdaptiveCards/issues/3723

## Description
The `<video>` tag on mobile OSes (iOS and Android) behaves in a non-standard way. In the context of this issue:
- Non-muted videos will not auto play, even if the `autoplay` attribute is specified
- The JS API `videoTag.play()` will not work if not triggered via explicit user interaction
- iOS only: The `playsinline` attribute must be specified on the `<video>` tag, otherwise the video will only play full screen

Refer this this page for more details: https://webkit.org/blog/6784/new-video-policies-for-ios/

This fix detects if the renderer in running on a mobile OS and adds the appropriate attributes accordingly.

## How Verified
- Verified that the implementation still works as before in "old" Edge, "new" Edge, Chrome, Firefox, Opera using the visualizer
- Pending verification on iOS. Test at https://adaptivecardsci.z5.web.core.windows.net/pr/3732/

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/AdaptiveCards/pull/3732)

### CODE REVIEW

File: source/nodejs/adaptivecards/src/utils.ts [hunk 1]
```diff
@@ -4,6 +4,12 @@ import * as Enums from "./enums";
 import * as Shared from "./shared";
 import { HostConfig } from "./host-config";
 
+export function isMobileOS(): boolean {
+    let userAgent = window.navigator.userAgent;
+
+    return !!userAgent.match(/Android/i) || !!userAgent.match(/iPad/i) || !!userAgent.match(/iPhone/i);
```
Comment (shalinijoshi19) [inline source/nodejs/adaptivecards/src/utils.ts]: Are these the only 2 iOS device types we care about? (iPad/iPhone) ?

**Comment by shalinijoshi19** [comment]
@nesalang is there an existing scenario file that we could leverage for testing this (outside of the useragent requirement) today? If not @dclaux we should add the scenario file to a folder someplace in our samples to ensure we can subsequently test this for now on-demand for bug bashes for eg. Also is this something that we want to include in the upcoming v1.2.6 patch this month? If so do tag the issue with the label pls. Thanks!

File: source/nodejs/adaptivecards/src/utils.ts (hunk 1 above)
Comment (dclaux) [reply source/nodejs/adaptivecards/src/utils.ts]: There is only one more device, the iPod, and I don't think we really care.

Comment (matthidinger) [reply source/nodejs/adaptivecards/src/utils.ts]: This feels a bit crude of a check. Do we have a desire to behave differently if a user checked on "Show Desktop Version" or any other mobile-browser toggles?

Comment (dclaux) [reply source/nodejs/adaptivecards/src/utils.ts]: Why is it crude? It detects if the renderer is running on a mobile OS, in which case things must be done differently for autoplay to work.

**Comment by nesalang** [comment]
yes, we do have a existing test file. 

---
In reply to: [575829211](https://github.com/microsoft/AdaptiveCards/pull/3732#issuecomment-575829211) [](ancestors = 575829211)

**Pull request merged**